src/org/sosy_lab/java_smt/solvers/z3/Z3OptimizationProver.java

```diff
@@ -197,8 +197,12 @@ private Optional<Rational> round(int handle, Rational epsilon, RoundingFunction
   }
 
   @Override
-  protected long getZ3Model() {
-    return Native.optimizeGetModel(z3context, z3optSolver);
+  protected long getZ3Model() throws InterruptedException {
+    try {
+      return Native.optimizeGetModel(z3context, z3optSolver);
+    } catch (Z3Exception e) {
+      throw new InterruptedException("Z3 model generation interrupted.");
+    }
```

Comment:
This code is wrong. Z3Exceptions can have other causes as well. Please use proper handling from Z3FormulaCreator.handleZ3Exception(e).

Reply:
fixed